src/citrine/resources/ara_job.py

```diff
@@ -0,0 +1,171 @@
+    status = properties.String("status")
+    dependencies = PropertySet(String(), "dependencies")
+    failure_reason = properties.Optional(String(), "failure_reason")
+
+    def __init__(
+            self,
+            id: str,
+            task_type: str,
+            status: str,
+            dependencies: Set[str],
+            failure_reason: Optional[str]
+    ):
+        self.id = id
+        self.task_type = task_type
+        self.status = status
+        self.dependencies = dependencies
+        self.failure_reason = failure_reason
+
+
+class TaskNodeType(Property[TaskNode, dict]):
+    @property
+    def underlying_types(self):
+        return TaskNode
+
+    @property
+    def serialized_types(self):
+        return dict
+
+    def _serialize(self, value: TaskNode) -> dict:
+        result = {"id": value.id, "task_type": value.task_type, "status": value.status,
+                  "failure_reason": value.failure_reason, "dependencies": value.dependencies}
+        return result
+
+    def _deserialize(self, value: dict) -> TaskNode:
+        return TaskNode.build(value)
+
+
+class JobStatusResponse(Resource['JobStatusResponse']):
+    """
+    [ALPHA] a response to a job status check
+
+    Parameters
+    ----------
+    job_type: str
+        the type of job for this status report
+    status: str
+        the actual status of the job
+    tasks: List[TaskNode]
+        all of the constituent task required to complete this job
+    """
+
+    job_type = properties.String("job_type")
+    status = properties.String("status")
+    tasks = properties.List(TaskNodeType, "tasks")
+
+    def __init__(
+            self,
+            job_type: str,
+            status: str,
+            tasks: List[TaskNode]
+    ):
+        self.job_type = job_type
+        self.status = status
+        self.tasks = tasks
+
+
+class AraJobFramework(Resource["AraJobFramework"]):
+    """
+    [ALPHA] a resource for submitting a job and getting job status
+
+    Parameters
+    ----------
+    session: Session
+        the remote session for hitting framework endpoints
+    """
+    _path_template = 'projects/{project_id}'
+
+    def __init__(self, session: Session):
+        self.session = session
+
+    def build_ara_table(self, project: Project,
+                        ara_def: AraDefinition) -> Union[ApiError, AraJobSubmissionResponse]:
+        """
+        [ALPHA] submit an ara table construction job
+
+        This method makes a call out to the job framework to start a new job to build
+        the respective table for a given AraDefinition.
+
+        Parameters
+        ----------
+        project: Project
+            The project on behalf of which the job executes.
+        ara_def: AraDefinition
+            the ara definition describing the new table
+        """
+        job_id = uuid4()
+        url_suffix: str = "/ara-definitions/{ara_definition}/versions/{version_number}/build?job_id={job_id}"
```

Comment:
there are arguments to `post_resource` for specifying query params, e.g. https://github.com/CitrineInformatics/citrine-python/blob/master/src/citrine/resources/data_concepts.py#L389